builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,326 @@
+	} else {
+		d.Set("arn", arn)
+		resp, err := conn.ListTagsForResource(&rds.ListTagsForResourceInput{
+			ResourceName: aws.String(arn),
+		})
+
+		if err != nil {
+			log.Printf("[DEBUG] Error retrieving tags for ARN: %s", arn)
+		}
+
+		var dt []*rds.Tag
+		if len(resp.TagList) > 0 {
+			dt = resp.TagList
+		}
+		d.Set("tags", tagsToMapRDS(dt))
+	}
+
+	return nil
+}
+
+func resourceAwsDbEventSubscriptionRetrieve(
+	d *schema.ResourceData, rdsconn *rds.RDS) (*rds.EventSubscription, error) {
+
+	name := d.Get("name").(string)
+	request := &rds.DescribeEventSubscriptionsInput{
+		SubscriptionName: aws.String(name),
+	}
+
+	describeResp, err := rdsconn.DescribeEventSubscriptions(request)
+	if err != nil {
+		if rdserr, ok := err.(awserr.Error); ok && rdserr.Code() == "SubscriptionNotFound" {
+			log.Printf("[WARN] No RDS Event Subscription by name (%s) found", name)
+			d.SetId("")
+			return nil, nil
+		}
+		return nil, fmt.Errorf("Error reading RDS Event Subscription %s: %s", name, err)
+	}
+
+	if len(describeResp.EventSubscriptionsList) != 1 {
+		return nil, fmt.Errorf("Unable to find RDS Event Subscription: %#v", describeResp.EventSubscriptionsList)
+	}
+
+	return describeResp.EventSubscriptionsList[0], nil
+}
+
+func resourceAwsDbEventSubscriptionUpdate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+
+	d.Partial(true)
+
+	// check/update tags, missing from go-aws-sdk?
+	if d.HasChange("enabled") || d.HasChange("event_categories") ||
+		d.HasChange("sns_topic") || d.HasChange("source_type") {
+
+		eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+		eventCategories := make([]*string, eventCategoriesSet.Len())
+		for _, eventCategory := range eventCategoriesSet.List() {
+			eventCategories = append(eventCategories, aws.String(eventCategory.(string)))
+		}
+
+		_, err := rdsconn.ModifyEventSubscription(&rds.ModifyEventSubscriptionInput{
+			SubscriptionName: aws.String(d.Id()),
+			Enabled:          aws.Bool(d.Get("enabled").(bool)),
+			SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+			SourceType:       aws.String(d.Get("source_type").(string)),
+			EventCategories:  eventCategories,
+		})
+
+		if err != nil {
+			return fmt.Errorf("Modifying RDS Event Subscription %s failed: %s", d.Id(), err)
+		}
+
+		log.Println(
+			"[INFO] Waiting for RDS Event Subscription modification to finish")
+
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"modifying"},
+			Target:     []string{"active"},
+			Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta.(*AWSClient).rdsconn),
+			Timeout:    40 * time.Minute,
+			MinTimeout: 10 * time.Second,
+			Delay:      30 * time.Second, // Wait 30 secs before starting
+		}
+
+		// Wait, catching any errors
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf("Modifying RDS Event Subscription %s failed: %s", d.Id(), err)
+		}
```

Comment:
We should also be calling `setPartial` for all fields that have been successfully modified here - i.e. `enabled`, `event_categories`, `sns_topic` and `source_type`, otherwise if an error occurs when setting tags below, these changes might be lost, or rather treated as unsaved.